src/node.cc

```diff
@@ -3746,6 +3746,7 @@ static void CheckIfAllowedInEnv(const char* exe, bool is_env,
     "--throw-deprecation",
     "--no-warnings",
     "--napi-modules",
+    "--abort-on-uncaught-exception",
```

Comment:
V8 options can use both `-` and `_` (https://nodejs.org/api/cli.html#cli_v8_options), so we should cover the variants (same for `--max_old_space_size`).